### ISSUE TITLE
About the data in your paper.

### ISSUE THREAD

**Issue by xnuohz** [ysig/GraKeL#17]
I used GraKel on 3 graph classiﬁcation datasets (MUTAG, ENZYMES, NCI1), it seems that my results are far away from your paper. I dont't know the reason. Could you explain this? Btw, PTC-MR is currently unsupported. And my source code is here: https://github.com/xnuohz/graph-kernel. Thanks:)

**Comment by ysig** [comment]
Hi,

thanks a lot for your interest in the library.
Let me know a few information:
- which paper (which one)?
- what do you mean is not supported?
- I think I haven't executed core-framework with graphlet, in any experiments I did?
- How do you evaluate your results?
- Do you do cross-validation on parameters - what parameters did you chose?

**Comment by xnuohz** [comment]
1. the paper is "A Degeneracy Framework for Graph Similarity".
2. I used the API[fetch_dataset] to load dataset[PTC-MR] from local path, and got the error "not supported".
3. I evaluated the data by 10-fold cross-validation on 6 kernels, and the result is the mean of this 10 folds.
4. my grakel version is "grakel-dev 0.1a5".

The code for bioinformatics datasets is as follows.
```python
#!/usr/bin/env python3
# -*- coding:utf-8 -*-
import time
import numpy as np
from sklearn.metrics import accuracy_score
from sklearn.model_selection import KFold
from sklearn import svm
from grakel import datasets
from grakel import GraphKernel


kernels = {
    "Shortest Path": [{"name": "shortest_path"}],

    "Graphlet Sampling": [{"name": "graphlet_sampling",
                           "sampling": {"n_samples": 500}}],

    "Weisfeiler-Lehman/Subtree": [{"name": "weisfeiler_lehman", "niter": 5},
                                  {"name": "subtree_wl"}],

    "Pyramid match": [{"name": "pyramid_match", "d": 6, "L": 4}],

    "Core Shortest Path": [{"name": "core_framework"},
                           {"name": "shortest_path"}],

    "Core Graphlet Sampling": [{"name": "core_framework"},
                               {"name": "graphlet_sampling", "sampling": {"n_samples": 500}}],

    "Core Weisfeiler-Lehman/Subtree": [{"name": "core_framework"},
                                       {"name": "weisfeiler_lehman", "niter": 5},
                                       {"name": "subtree_wl"}],

    "Core Pyramid match": [{"name": "core_framework"},
                           {"name": "pyramid_match", "d": 6, "L": 4}]
}

Datasets = ["MUTAG", "ENZYMES", "NCI1", "PTC-MR", "DD"]
Methods = sorted(list(kernels.keys()))

for j, d in enumerate(Datasets):
    print(d)
    dataset_d = datasets.fetch_dataset(d, verbose=False, data_home="../dataset")
    G, y = np.asarray(dataset_d.data), np.asarray(dataset_d.target)

    stats = {m: {"acc": list(), "time": list()} for m in Methods}

    kfold = KFold(n_splits=10, random_state=1, shuffle=True)

    for train_idx, test_idx in kfold.split(G, y):
        train_g, train_y = G[train_idx], y[train_idx]
        test_g, test_y = G[test_idx], y[test_idx]

        for i, k in enumerate(Methods):
            gk = GraphKernel(kernel=kernels[k], normalize=True)

            start = time.time()
            k_train = gk.fit_transform(train_g)
            k_test = gk.transform(test_g)
            end = time.time()

            clf = svm.SVC(kernel='precomputed')
            clf.fit(k_train, train_y)

            pred_y = clf.predict(k_test)

            stats[k]["acc"].append(accuracy_score(test_y, pred_y))
            stats[k]["time"].append(end - start)

    for m in Methods:
        print("kernel: ", m,
              "time: ", np.round(np.mean(stats[m]["time"]), 2), "~", np.round(np.std(stats[m]["time"]), 2),
              "acc: ", np.round(np.mean(stats[m]["acc"]), 2), "~", np.round(np.std(stats[m]["acc"]), 2))
```

**Comment by giannisnik** [comment]
Dear @xnuohz ,

Thank you for your interest in our work.

- The library downloads the datasets from this repository: https://ls11-www.cs.tu-dortmund.de/staff/morris/graphkerneldatasets. The name of the PTC Male Rats dataset in that repository is PTC**_**MR and not PTC-MR. This is why you get the "not supported" error.

- With regards to the performance of the kernels, note that we have optimized the hyperparameters of the kernels based on a 90%-10% split of the training set of each fold (see subsection 4.2 in the paper) and we have also optimized the hyperparameter C of the SVM classifier. Instead, in your code, you use the default hyperparameters for the kernels and you do not optimize the value of C (i.e., you set it equal to 1). I guess you will manage to reproduce the results if you also optimize these hyperparameters.

Hope this helps!!
